orttraining/orttraining/core/agent/training_agent.cc

```diff
@@ -50,25 +50,26 @@ TrainingAgent::TrainingAgent(InferenceSession& session,
 TrainingAgent::~TrainingAgent() = default;
 
 common::Status TrainingAgent::RunForward(const std::vector<OrtValue>& feeds, std::vector<OrtValue>& fetches,
-                                         PartialGraphExecutionState& state) {
+                                         PartialGraphExecutionState& state, const OrtValueCache& cache) {
```

Comment:
> const OrtValueCache& cache)  [](http://example.com/codeflow?start=75&length=29)

It seems we will now always create a cache but this will add additional overhead in the case when gradient accumalation is turned off. I think this is in a hot path, have you benchmarked its overhead or considered not creating an empty cache?

Reply:
see https://github.com/microsoft/onnxruntime/pull/8273#issuecomment-900515827